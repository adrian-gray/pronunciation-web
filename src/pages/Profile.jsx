import React from 'react'
import { Redirect } from 'react-router-dom'
import firebase from 'firebase/app'

import {
  Paper,
  Typography,
  withStyles
} from '@material-ui/core'

const styles = (theme) => ({
  page: theme.page,
  headspace: theme.headspace
})

class Profile extends React.Component {
  constructor (props) {
    super(props)

    this.state = {
      displayName: null,
      photoURL: null,
      subscriber: false
    }
  }

  componentDidMount () {
    firebase.auth().onAuthStateChanged(user => {
      this.setState({
        displayName: user.displayName,
        photoURL: user.photoURL
      })
    })
  }

  render () {
    const { classes } = this.props

    if (!this.props.isLoggedIn) {
      return <Redirect to='/' />
    }

    return (
      <Paper className={classes.page}>
        <div className={classes.headspace}>
          <Typography variant='title' gutterBottom>
            {`Your profile`}
          </Typography>
          <Typography>
            {this.state.displayName}
          </Typography>
          <div>
            <img width='100px' src={this.state.photoURL} />
          </div>
        </div>
      </Paper>
    )
  }
}

export default withStyles(styles)(Profile)
