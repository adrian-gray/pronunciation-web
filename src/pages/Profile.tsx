import React from "react";
import { Redirect } from "react-router-dom";

import Container from "react-bootstrap/Container";

export default props => {
  const { user } = props;

  if (!user) {
    return <Redirect to="/" />;
  }

  return (
    <Container className="page">
      <div className="head-space">
        <h3>{`Your profile`}</h3>
        <p>{user.displayName}</p>
        <div>
          <img width="100px" src={user.photoURL} />
        </div>
      </div>
    </Container>
  );
};
