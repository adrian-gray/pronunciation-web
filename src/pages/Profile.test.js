import React from "react";
import { BrowserRouter } from "react-router-dom";

import { render, cleanup } from "@testing-library/react";

import Profile from "./Profile";

afterEach(cleanup);

test("Profile", async () => {
  const props = {
    user: {
      displayName: "Bob",
      photoURL: "foo.com/biteMe.jpg"
    }
  };

  const { container } = render(
    <BrowserRouter>
      <Profile {...props} />
    </BrowserRouter>
  );

  expect(container).toMatchSnapshot();
});
