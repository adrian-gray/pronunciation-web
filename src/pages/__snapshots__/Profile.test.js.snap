// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile 1`] = `
<div>
  <div
    class="page"
  >
    <div
      class="head-space"
    >
      <h3>
        Your profile
      </h3>
      <p>
        Bob
      </p>
      <div>
        <img
          src="foo.com/biteMe.jpg"
          width="100px"
        />
      </div>
    </div>
  </div>
</div>
`;
